src/Deals.php

```diff
@@ -101,13 +101,23 @@ public function getBySearch($search, $limit = false, $offset = 0)
 		return $this->formatDealsForReturn($deals);
 	}
 
-	public function getBaseQuery($limit = false, $offset = 0)
+	public function getBaseQuery($limit = false, $offset = 0, $orderField = 'fRating', $desc = true)
 	{
 		$query = DB::table('deals')
 					->select(DB::raw('fmtc_merchants.*, fmtc_deals.*'))
 					->join('merchants', 'deals.nMerchantID', '=', 'merchants.nMerchantID')
-					->where('dtEndDate', '>=', date('Y-m-d H:i:s'))
-					->orderBy('fRating', 'desc');
+					->where('dtEndDate', '>=', date('Y-m-d H:i:s'));
+		
+		$order = 'asc';
+		if($desc) {
+			$order = 'desc';
+		}
+		
+		$query->orderBy($orderField, $order);
+
+		if($orderField !== 'fRating') {
+			$query->orderBy('fRating', 'desc');
+		}
```

Comment:
if the $orderField is not set to "fRating" it looks like it is ordered by twice. It also seems that the only thing you can order by is "fRating" given this flow. the default is already set in the parameters. Why not just remove this whole if statement?